src/promptflow/promptflow/executor/flow_validator.py

```diff
@@ -33,11 +33,12 @@ def _ensure_nodes_order(flow: Flow):
                 if i.value_type != InputValueType.NODE_REFERENCE:
                     continue
                 if i.value not in dependencies:
-                    msg = (
-                        f"Node '{n.name}' references a non-existent node '{i.value}' in your flow. "
-                        f"Please review your flow to ensure that the node name is accurately specified."
+                    msg_format = (
+                        "Flow is defined incorrectly. Node '{node_name}' references a non-existent node "
+                        "'{ref_node_name}' in your flow. Please review your flow to ensure that the "
+                        "node name is accurately specified."
                     )
-                    raise NodeReferenceNotFound(message=msg)
+                    raise NodeReferenceNotFound(message_format=msg_format, node_name=n.name, ref_node_name=i.value)
```

Comment:
You can define `message_format` directly in the error instead of `message_format=msg_format`.
```suggestion
                    raise NodeReferenceNotFound(
                        message_format=(
                            "Flow is defined incorrectly. Node '{node_name}' references a non-existent node "
                            "'{ref_node_name}' in your flow. Please review your flow to ensure that the "
                            "node name is accurately specified."
                        ), 
                        node_name=n.name, 
                        ref_node_name=i.value
                    )
```

Reply:
ref_node_name -> reference_node_name?